internal/controller/controllers/infraenv_controller.go

```diff
@@ -183,10 +184,46 @@ func (r *InfraEnvReconciler) updateInfraEnv(ctx context.Context, log logrus.Fiel
 
 	updateParams.InfraEnvUpdateParams.ImageType = r.Config.ImageType
 
+	existingKargs, err := existingDiscoveryKernelArguments(internalInfraEnv)
+	if err != nil {
+		return nil, err
+	}
+	if !funk.Equal(infraEnv.Spec.DiscoveryKernelArguments, existingKargs) {
```

Comment:
Does order matter or should this do some kind of set comparison?

Reply:
Seems to me yes.  That is the order that will be applied to iPXE and ISO.  Do you think differently?

Reply:
No, I'm actually asking if the order matters when you provide kernel arguments.

If it doesn't then we might be able to skip some unnecessary updates, but if it does then this looks fine to me.
I suppose keeping it as is is safer in case order does indeed matter :man_shrugging: